helm-chart/todoapp/templates/deployment.yml

```diff
@@ -0,0 +1,87 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.Release.Name}}-{{.Chart.Name}}
+  namespace: {{.Values.todoapp.namespace}}
+spec:
+  replicas: 1
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: {{ .Values.todoapp.rollingUpdate.maxSurge }}
+      maxUnavailable: {{ .Values.todoapp.rollingUpdate.maxUnavailable}}
+  selector:
+    matchLabels:
+      app: {{.Chart.Name}}
+  template:
+    metadata:
+      labels:
+        app: {{.Chart.Name}}
+    spec:
+      serviceAccountName: {{ .Values.todoapp.RBAC.name}}
+      containers:
+      - name: {{.Values.todoapp.image.name}}
+        image: {{.Values.todoapp.image.repository}}:{{.Values.todoapp.image.version}}
+        volumeMounts:
+        - name: data
+          mountPath: /app/data
+        - name: app-secrets-volume
+          mountPath: "/app/secrets"
+          readOnly: true
+        - name: app-config-volume
+          mountPath: "/app/configs"
+          readOnly: true
+        resources:
+          requests:
+            memory: {{.Values.todoapp.resources.requests.memory}}
+            cpu: {{.Values.todoapp.resources.requests.cpu}}
+          limits:
+            memory: {{.Values.todoapp.resources.limits.memory}}
+            cpu: {{.Values.todoapp.resources.limits.cpu}}
+        env:
+        - name: PYTHONUNBUFFERED
+          valueFrom:
+            configMapKeyRef:
+              name: {{ .Chart.Name }}-config
+              key: PYTHONUNBUFFERED
+        {{- range $key, $value := .Values.todoapp.secret }}
+        - name: {{ $key }}
+          valueFrom:
+            secretKeyRef:
+              name: {{ $.Release.Name }}-secret
+              key: {{ $key }}
+        {{ end }}
+        ports:
+        - containerPort: 8080
+        livenessProbe:
+          httpGet:
+            path: api/health
+            port: 8080
+          initialDelaySeconds: 60
+          periodSeconds: 5
+        readinessProbe:
+          httpGet:
+            path: api/ready
+            port: 8080
+          initialDelaySeconds: 5
+          periodSeconds: 5
+      volumes:
+      - name: data
+        persistentVolumeClaim:
+          claimName: pvc-data
+      - name: app-secrets-volume
+        secret:
+          secretName: app-secret
+      - name: app-config-volume
+        configMap:
+          name: app-config
+      affinity:
+        nodeAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: {{ .Values.todoapp.nodeAffinity.weight }}
+            preference:
+              matchExpressions:
+              - key: {{ .Values.todoapp.nodeAffinity.key }}
+                operator: {{ .Values.todoapp.nodeAffinity.operator }}
+                values:
+                - {{ .Values.todoapp.nodeAffinity.values }}
```

Comment:
Verify that the node affinity values (`weight`, `key`, `operator`, `values`) are correctly defined in your `values.yaml` file. Incorrect node affinity settings could affect pod scheduling.